storage/block/types.go

```diff
@@ -317,12 +315,24 @@ type Options interface {
 	// ClockOptions returns the clock options
 	ClockOptions() clock.Options
 
-	// SetDatabaseBlockAllocSize sets the databaseBlockAllocSize
+	// SetInstrumentOptions sets the instrument options
+	SetInstrumentOptions(value instrument.Options) Options
+
+	// InstrumentOptions returns the instrument options
+	InstrumentOptions() instrument.Options
+
+	// SetDatabaseBlockAllocSize sets the database block alloc size
 	SetDatabaseBlockAllocSize(value int) Options
 
-	// DatabaseBlockAllocSize returns the databaseBlockAllocSize
+	// DatabaseBlockAllocSize returns the database block alloc size
 	DatabaseBlockAllocSize() int
 
+	// SetWiredList sets the database block wired list
+	SetWiredList(value *WiredList) Options
+
+	// WiredList returns the database block wired list
+	WiredList() *WiredList
```

Comment:
Good question, so this is definitely valid - that is why I made the API to update WiredList completely asynchronous.  You simply enqueue an update to the wired list and a background goroutine reads the update and updates the list.  Enqueuing to a channel can be contentions even itself however, so this enqueuing is done asynchronously after a read completes with the onStreamDoneFinalizer on the block.

This avoids the central contention and allows reads to still complete quickly.